internal/backoff/backoff_test.go

```diff
@@ -19,18 +19,183 @@ package backoff
 
 import (
 	"context"
-	"errors"
+	"fmt"
+	"reflect"
 	"testing"
+	"time"
 
-	"github.com/vdaas/vald/internal/log"
+	"github.com/vdaas/vald/internal/errors"
 )
 
-func TestNextDuration(t *testing.T) {
-	log.Init(log.DefaultGlg())
-	t.Run("Text NextDuration", func(t *testing.T) {
-		bo := New()
-		bo.Do(context.Background(), func() (interface{}, error) {
-			return nil, errors.New("what a fuck")
+func TestNew(t *testing.T) {
+	type args struct {
+		opts []Option
+	}
+
+	type test struct {
+		name      string
+		args      args
+		want      *backoff
+		checkFunc func(got, want *backoff) error
+	}
+
+	tests := []test{
+		{
+			name: "initialize",
+			want: &backoff{
+				initialDuration:  float64(10 * time.Millisecond),
+				backoffTimeLimit: 5 * time.Minute,
+				maxDuration:      float64(time.Hour),
+				jitterLimit:      float64(time.Minute),
+				backoffFactor:    1.5,
+				maxRetryCount:    50,
+				errLog:           true,
+				durationLimit:    float64(time.Hour) / 1.5,
+			},
+			checkFunc: func(got *backoff, want *backoff) error {
+				got.jittedInitialDuration, want.jittedInitialDuration = 1, 1
```

Comment:
Because `jittedInitialDuration` is generated from random numbers, it is difficult to compare, so ignore it